### PR TITLE
[SwiftLint] SwiftLint compatible

### PR DESCRIPTION
**Pull request by wongzigii**
Use swiftlint to format codes and codeblocks in Sources and Tests. Related to #641

### CODE REVIEW

**Comment by wongzigii** [comment]
When running swiftlint locally, I got 26 warnings about `Force Unwrapping Violation`.  Should we also reslove them? 

````
Linting 'ItTests.swift' (39/42)
/Users/Wongzigii/Quick/Tests/QuickTests/QuickTests/FunctionalTests/ItTests.swift:11:35: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
/Users/Wongzigii/Quick/Tests/QuickTests/QuickTests/FunctionalTests/ItTests.swift:16:35: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
/Users/Wongzigii/Quick/Tests/QuickTests/QuickTests/FunctionalTests/ItTests.swift:82:37: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
/Users/Wongzigii/Quick/Tests/QuickTests/QuickTests/FunctionalTests/ItTests.swift:93:41: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
/Users/Wongzigii/Quick/Tests/QuickTests/QuickTests/FunctionalTests/ItTests.swift:6:14: warning: Function Body Length Violation: Function body should span 40 lines or less excluding comments and whitespace: currently spans 62 lines (function_body_length)
Linting 'PendingTests.swift' (40/42)
/Users/Wongzigii/Quick/Tests/QuickTests/QuickTests/FunctionalTests/PendingTests.swift:38:29: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
/Users/Wongzigii/Quick/Tests/QuickTests/QuickTests/FunctionalTests/PendingTests.swift:6:1: warning: Variable Name Violation: Variable name should be between 3 and 40 characters long: 'onlyPendingExamplesBeforeEachExecutedCount' (variable_name)
Linting 'SharedExamples+BeforeEachTests.swift' (41/42)
Linting 'SharedExamplesTests.swift' (42/42)
/Users/Wongzigii/Quick/Tests/QuickTests/QuickTests/FunctionalTests/SharedExamplesTests.swift:46:25: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
/Users/Wongzigii/Quick/Tests/QuickTests/QuickTests/FunctionalTests/SharedExamplesTests.swift:47:30: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
/Users/Wongzigii/Quick/Tests/QuickTests/QuickTests/FunctionalTests/SharedExamplesTests.swift:52:25: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
Done linting! Found 26 violations, 3 serious in 42 files.
````

**Comment by ikesyo** [comment]
> b6db181

SwiftLint does not support Linux yet: https://github.com/realm/SwiftLint/issues/732

**Comment by wongzigii** [comment]
@ikesyo 😢  Alternatively, maybe we can move swiftlint to osx matrix.

**Comment by ikesyo** [comment]
FYI Hound supports SwiftLint: https://houndci.com/

**Comment by MP0w** [comment]
@wongzigii punting only in one job is fine since the output would always be the same. Same for Danger. BTW Danger has a plugin for swiftlint, if we move it to the Dangerfile then you would also get report by your DangerBot

**Comment by wongzigii** [comment]
@MP0w Totally agree! Run script can ensure new contributor to format their code.

**Comment by jeffh** [comment]
Would the run script require users (Carthage or sub module) to also have swiftlint installed to build Quick?

**Comment by MP0w** [comment]
@jeffh no, they would just get a warning https://github.com/realm/SwiftLint/blob/master/README.md#usage

File: Tests/QuickTests/QuickFocusedTests/FocusedTests.swift [hunk 1]
```diff
@@ -21,7 +21,6 @@ class FunctionalTests_FocusedSpec_Focused: QuickSpec {
             fit("has a focused example that passes (3)") {}
         }
 
-        // TODO: Port fitBehavesLike to Swift.
```
Comment (wongzigii) [inline Tests/QuickTests/QuickFocusedTests/FocusedTests.swift]: I removed this TODO to avoid todo violation. Open a new issue if need.

**Comment by wongzigii** [comment]
Please review @Quick/core

**Comment by istx25** [comment]
LGTM. I'd like a few others to review as well (if people would like to) before we merge. But otherwise, I'm happy to have this merged in. :)

File: Tests/QuickTests/QuickFocusedTests/FocusedTests.swift (hunk 1 above)
Comment (modocache) [inline Tests/QuickTests/QuickFocusedTests/FocusedTests.swift]: Good point! Done: https://github.com/Quick/Quick/issues/643

File: .swiftlint.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+disabled_rules:
+  - line_length
+  - type_name
+  - valid_docs
+  - empty_count
+  - force_unwrapping
+  - function_body_length
+  - variable_name
```
Comment (modocache) [inline .swiftlint.yml]: It'd be awesome if Quick contributors could send pull requests to un-disable each of these rules. One pull request per rule would be great. :)

**Comment by modocache** [comment]
Awesome. Thanks everyone!

File: .swiftlint.yml (hunk 2 above)
Comment (jeffh) [inline .swiftlint.yml]: I've updated the original issue #641

Comment (modocache) [inline .swiftlint.yml]: Great, thanks! 🙇

**Pull request merged**